packages/jaeger-ui/src/middlewares/index.js

```diff
@@ -45,4 +47,29 @@ export const historyUpdateMiddleware = store => next => action => {
   return next(action);
 };
 
+export const transformOTLPMiddleware = store => next => action => {
+  if (action.type === String([`${loadJsonTraces}_FULFILLED`])) {
```

Comment:
can you explain what's happening here?

Reply:
Im checking if the actions thats being invoked is loadJsonTrace_FULLFILLED (which is what is called after the file upload is finished) After which I check if its a OTLP type trace and make an API call. 

If an API call was made I make a custom action and pass it on the same reducer using the `next` variable(its a function) If not it just forwards the actioon to the default reducer

Reply:
> if the actions thats being invoked is `loadJsonTrace_FULLFILLED`

but why is it `${loadJsonTraces}` instead? Is `loadJsonTraces` a constant somewhere?

Reply:
Yes it's a constant defined in file-upload-api.tsx (not sure about this file name)

Reply:
but it's not a string constant:
```
export const loadJsonTraces = createAction(
  '@FILE_READER_API/LOAD_JSON',
  fileList => readJsonFile(fileList),
  fileList => ({ fileList })
);
```
yet you're using it as if it's a string or renders into a string.

Reply:
from redux code : 

https://github.com/reduxjs/redux-toolkit/blob/261ff262745ff894c108944c5cdc7f49c6d61214/src/createAction.ts#L261-L281

And : 
https://stackoverflow.com/questions/61223096/also-the-action-creator-overrides-tostring-so-that-the-action-type-becomes-it
`createAction()` overrides the functions toString method (which is what is called during `${}`